### DISCUSSION TITLE
Multiple additional SSH keys with labels on re-apply

### DISCUSSION THREAD

**Discussion by MariusSchiffer** [mysticaltech/terraform-hcloud-kube-hetzner#1129] [Ideas]
### Description

I set up multiple SSH keys with label `role=admin`. I'm expecting terraform to roll out any and all SSH keys that have this label.
When I'm adding a new SSH key, it doesn't seem to apply it.

### Kube.tf file

```terraform
hcloud_ssh_key_id = "schiffer" 
  # * Your private key must be "ssh_private_key = null" when you want to use ssh-agent for a Yubikey-like device authentification or an SSH key-pair with a passphrase.
  # For more details on SSH see https://github.com/kube-hetzner/kube-hetzner/blob/master/docs/ssh.md
  # You can add additional SSH public Keys to grant other team members root access to your cluster nodes.
  # ssh_additional_public_keys = []

  # You can also add additional SSH public Keys which are saved in the hetzner cloud by a label.
  # See https://docs.hetzner.cloud/#label-selector
  ssh_hcloud_key_label = "role=admin"
```

It does seem to refresh `module.kube-hetzner.data.hcloud_ssh_keys.keys_by_selector[0]`.

So to reproduce. create a cluster including a hcloud_key_label key. Add a key with the same label and re-apply. Infrastructure does not change. It should change the cloudinit config 

### Platform

Linux

**Comment by mysticaltech** [comment]
@MariusSchiffer This should work for first deploy only. We do not yet have support to apply it at a later stage on re-apply. We do support this kind of behavior for other things. If you want to add support for this, PR is most welcome. Usually terraform is really just for deploy, so that's more like custom behavior and extra sugar, but I can see the need for it.
